### PR TITLE
Support for ROS2 standalone

### PR DESCRIPTION
**Pull request by msz-rai**

### CODE REVIEW

File: docs/Ros2Extension.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+1. Run `x64 Native Tools Command Prompt for VS 2019` and navigate to RGL repository.
+2. Run `python setup.py --with-ros2-standalone` command to build RGL with ROS2 extension and install ROS2 libraries.
+3. Copy all ROS2 libraries from `<build-dir>/ros2_standalone/` into `RobotecGPULidar.dll` location, or extend environment variable `Path` appropriately.
+
+#### Tips for integrating RGL + ROS2 standalone with Unity and [ROS2ForUnity](https://github.com/RobotecAI/ros2-for-unity) plugin.
+1. Build RGL with ROS2 standalone as described above.
+2. Copy `<build-dir>/RobotecGPULidar.dll` and all depend libraries located in `<build-dir>` to appropriate directory in your RGLUnityPlugin.
+3. Copy all ROS2 libraries from `<build-dir>/ros2_standalone/` to `Ros2ForUnity\Plugins\Windows\x86_64` directory. Skip for duplicates.
+
+In this case, RGL's ROS2 standalone build is dependent on ROS2ForUnity's ROS2 standalone build. RobotecGPULidar library will find ROS2 because ROS2ForUnity sets environment variable `Path` for the ROS2 libraries.
```
Comment (prybicki) [inline docs/Ros2Extension.md:92]: Does it mean that RGL is always loaded after R2FU? How is the order handled?

**Review by prybicki — COMMENTED** [review]
Great job!
- Added a few suggestions and questions to address.
- Proposed changes to setup.py in a separate PR to this PR.

File: docs/Ros2Extension.md (hunk 1 above)
Comment (msz-rai) [reply docs/Ros2Extension.md:93]: R2FU is initialized before any `Awake()` or `Start()`. This is achieved by [RuntimeInitializeOnLoadMethod](https://docs.unity3d.com/ScriptReference/RuntimeInitializeOnLoadMethodAttribute.html) attribute. See [source code](https://github.com/tier4/AWSIM/blob/main/Assets/AWSIM/Scripts/ROS/SimulatorROS2Node.cs#L20).

File: docs/Ros2Extension.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+#### Steps for standalone
+1. Use script `setup.py` with option `--with-ros2-standalone`
+    - You can specify run-time search path(s) for RGL library by adding option `--lib-rpath <paths>`.
+2. Copy all ROS2 libraries from `<build-dir>/ros2_standalone/` to `libRobotecGPULidar.so` location (or location defined with `--lib-rpath`).
+
+#### Tips for integrating RGL + ROS2 standalone with Unity and [ROS2ForUnity](https://github.com/RobotecAI/ros2-for-unity) plugin.
+1. Build RGL with command:
+    ```bash
+    ./setup.py --with-ros2-standalone --lib-rpath \$ORIGIN/.plugin --make="-j"
```
[the remark this answers is not in the document]
Comment (msz-rai) [reply docs/Ros2Extension.md:46]: I have added an explanation.

**Review by prybicki — APPROVED** [review]

**Pull request merged**